### PR TITLE
feat: Add a way to to obtain information about the pinned widgets

### PR DESCRIPTION
**Pull request by qwadrox**
Would address 
https://github.com/ABausG/home_widget/issues/129
https://github.com/ABausG/home_widget/issues/223
https://github.com/ABausG/home_widget/issues/37

### CODE REVIEW

**Comment by qwadrox** [comment]
[ABausG](https://github.com/ABausG) Can you take a look? :)

File: example/lib/main.dart [hunk 1]
```diff
@@ -172,50 +172,87 @@ class _MyAppState extends State<MyApp> {
...
+            children: [
+              Text('Number of widgets: ${widgets.length}'),
+              const Divider(),
+              for (final widget in widgets)
+                Text(
+                  '- iOS Family: ${widget.family}, '
+                  'iOS Kind: ${widget.kind}, '
+                  'Android Widget id: ${widget.widgetId}, '
+                  'Android Class Name: ${widget.androidClassName}, '
+                  'Android Label: ${widget.label}',
```
Comment (ABausG) [inline example/lib/main.dart:194]: Maybe wrap the according lines with a `Platform` check to keep the info a bit smaller

**Review by ABausG — COMMENTED** [review]
Just a had a quick first glance. All in all looks already really good and will be super helpful!

Am I correct in assuming that getting the size of an installed widget is not possible on Android as the way I understand the API and your implementation of accessing the Info is that using the widgetIds of the Installed/Pinned Widgets it gets the general configuration of the widget so a min/max size would be possible but not the actual size the user is using? I guess that Information is not really relevant to users of home_widget as they are defining that. Do you have any idea/experience if it would be possible to get that information anyway? Ideally in this call?

**Comment by qwadrox** [comment]
> Just a had a quick first glance. All in all looks already really good and will be super helpful!
> 
> Am I correct in assuming that getting the size of an installed widget is not possible on Android as the way I understand the API and your implementation of accessing the Info is that using the widgetIds of the Installed/Pinned Widgets it gets the general configuration of the widget so a min/max size would be possible but not the actual size the user is using? I guess that Information is not really relevant to users of home_widget as they are defining that. Do you have any idea/experience if it would be possible to get that information anyway? Ideally in this call?

My original idea differed slightly, but you're grasping it well.

Here's what I know:


- The current approach, using `getAppWidgetInfo`, retrieves the AppWidgetProviderInfo, with its fields corresponding to those in the <appwidget-provider> XML tag. This means the values are predefined, providing enough information to differentiate each widget instance. Additionally, each instance has a unique **widgetId**, which can be useful for saving specific information to shared preferences, such as the layout used for the RemoteView or the values being displayed at a given time. So the **widgetId** offers flexibility in what information we can save.

- Moreover, it's possible to obtain the actual size in dips of the widgets by relying on `getAppWidgetOptions`. This size changes when adjusting the width or height. This feature can be useful for [size calculations](https://android.googlesource.com/platform/frameworks/base/+/HEAD/core/java/android/appwidget/AppWidgetHostView.java#402), and these properties are also settable. Although the "actual" size might be irrelevant in some cases, it can still be beneficial for certain applications (?) You can explore this functionality further in the comments on this branch: https://github.com/ABausG/home_widget/commit/8753880c4d66556bde2bd55df878a37ca53a911f. Feel free to try it out as well. 😄

- In my case it was irrelevant. Thats why I did not include these infos. 

- Installed/Pinned yes. Installed have a different meaning in android. Every widget on your phone even if you didn't place on home screen is installed, from all apps. That's why [getInstalledProviders](https://developer.android.com/reference/android/appwidget/AppWidgetManager#getInstalledProviders())() is not good for us. Maybe we should avoid using "installed" word. 

@ABausG

File: example/lib/main.dart (hunk 1 above)
Comment (qwadrox) [reply example/lib/main.dart]: [fe8af81](https://github.com/ABausG/home_widget/pull/233/commits/fe8af81aec8b1e9436ee5c37fb5a595ba0bc47b5)

**Review by ABausG — APPROVED** [review]
Works great!

I just refactored some variables in HomeWidgetInfo to include OS prefixes for all variables to make everything a bit clearer and indicate why certain things might be null.

Excited to fit this in to the next release!

**Pull request merged**